### PR TITLE
Added a describe_features function to the SASAFeaturizer.

### PR DESCRIPTION
**Pull request by dr-nate**
- [x] Implement feature / fix bug
 - [x] Add tests
 - [ ] Update changelog

Added a `describe_features` function to the SASAFeaturizer. Contingent on this PR to MDTraj also being accepted https://github.com/mdtraj/mdtraj/pull/1174. Not a huge addition but it's nice to have. Let me know if this is significant enough to add to the changelog. 

Also, I have tested it on real data, but didn't add any tests. Let me know if you want those too.

### CODE REVIEW

File: msmbuilder/featurizer/featurizer.py [hunk 1]
```diff
@@ -884,6 +884,55 @@ def __init__(self, mode='residue', **kwargs):
...
+        feature_descs = []
+        _, mapping = md.geometry.sasa.shrake_rupley(traj, mode=self.mode, get_mapping=True)
+        top = traj.topology
+
+        if self.mode == "residue":
+            resids = np.unique(mapping)
+            resseqs = [top.residue(ri).resSeq for ri in resids]
+            resnames = [top.residue(ri).name for ri in resids]
+            # Not sure if there's a better way to get the indices of atoms in a residue
+            atoms_in_res = [res.atoms for res in top.residues]
```
Comment (msultan) [inline msmbuilder/featurizer/featurizer.py]: This looks good to me and i cant think of a fancy list comprehensive way around this. Maybe @mpharrigan knows.

**Comment by msultan** [comment]
I think a test would still be nice. Take a look at some of the randomized tests that i wrote [here](https://github.com/msmbuilder/msmbuilder/blob/master/msmbuilder/tests/test_feature_descriptor.py) for taking a random frame and calculating a random feature index to make sure they match up.

File: msmbuilder/featurizer/featurizer.py [hunk 2]
```diff
@@ -884,6 +884,55 @@ def __init__(self, mode='residue', **kwargs):
...
+            aind_tuples = []
+            # For each resdiue...
+            for i,x in enumerate(atoms_in_res):
+                aind_tuples.append([])
+                # For each atom in the residues, append it's index
+                for y in x:
+                    aind_tuples[i].append(y.index)
```
Comment (msultan) [inline msmbuilder/featurizer/featurizer.py]: this could be 
```
aind_tuples.append([atom.index for atom in x])
```

**Comment by msultan** [comment]
Thanks for taking this on @dr-nate !

**Comment by dr-nate** [comment]
Ok. I didn't realize there was already test for the other `describe_features` functions. I've added them now and cleaned up the other code. If/when the MDTraj PR is accepted, this one can be accepted.

**Comment by msultan** [comment]
@mpharrigan could you take a look at the Travis error, I don't really understand whats going on here.

**Comment by mpharrigan** [comment]
Attempting to fix in https://github.com/msmbuilder/msmbuilder/pull/915

**Pull request merged**